pkg/products/rhssocommon/reconciler.go

```diff
@@ -100,9 +100,21 @@ func (r *Reconciler) CleanupKeycloakResources(ctx context.Context, inst *integre
 		Namespace: ns,
 	}
 
+	// check for CRD being created. If it is not, IsForbidden == false. If created - IsForbidden == true
+	keycloakCR := &keycloak.Keycloak{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      string(integreatlyv1alpha1.ProductRHSSO),
+			Namespace: inst.Namespace,
+		},
+	}
+	_, err := resources.Exists(ctx, serverClient, keycloakCR)
+	if err != nil && !k8serr.IsForbidden(err) {
+		return integreatlyv1alpha1.PhaseCompleted, nil
+	}
```

Comment:
Would be nice to have a unit test for this :thinking:

Reply:
Sorry, that it took me so long :sweat: . I've added unit tests and also refactored code to directly check the presence of CRD instead of relying on created CRs. Tested on the cluster - it works. Verification steps remain the same